### PR TITLE
Route Maps data model and examples for enterprise_sonic

### PR DESCRIPTION
**Pull request by kerry-meyer**
Note: Please see the order of examples listed at the bottom of the "sonic_route_maps.yml" file for the intended order of applying and reading the examples in this model.

### CODE REVIEW

File: models/enterprise_sonic/route_maps/sonic_route_maps.yml [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'community',
+      'license': 'Apache 2.0'
+    }
+NETWORK_OS: sonic
+RESOURCE: route_maps
+COPYRIGHT: Copyright 2019 Red Hat
```
Comment (stalabi1) [inline models/enterprise_sonic/route_maps/sonic_route_maps.yml:13]:
```suggestion
COPYRIGHT: Copyright 2023 Dell Inc. or its subsidiaries. All Rights Reserved
```

File: models/enterprise_sonic/route_maps/sonic_route_maps.yml [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+                - BGP Ethernet Virtual Private Network to be checked for
+                - a match with the target route
+              type: dict
+              required_one_of: [['default-route', 'route-type', 'vni']]
+              options:
+                default_route:
+                  description:
+                    - Default EVPN type-5 route
+                  type: bool
+                route-type:
```
Comment (stalabi1) [inline models/enterprise_sonic/route_maps/sonic_route_maps.yml:72]:
```suggestion
                route_type:
```
Hyphenated variable names aren't valid in playbooks, use underscore instead. Please correct all hyphenated variables in this file.

File: models/enterprise_sonic/route_maps/sonic_route_maps.yml [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+        sequence_num:
+          description:
+            - unique number in the range 1-66535 to specify priority of the map
+          type: int
+          required: true
+        match:
+          description: Criteria for matching the route map to a route
+          type: dict
+          options:
+            as-path:
```
Comment (stalabi1) [inline models/enterprise_sonic/route_maps/sonic_route_maps.yml:51]:
```suggestion
            as_path:
```

File: models/enterprise_sonic/route_maps/sonic_route_maps.yml (hunk 1 above)
Comment (kerry-meyer) [reply models/enterprise_sonic/route_maps/sonic_route_maps.yml]: Done.

Thanks for pointing this out.

File: models/enterprise_sonic/route_maps/sonic_route_maps.yml [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+  description:
+    - This module provides configuration management for route map parameters on devices running SONiC.
+  options:
+    config:
+      description:
+        - Specifies a list of route map configuration dictionaries
+      type: list
+      elements: dict
+      options:
+        name:
```
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply models/enterprise_sonic/route_maps/sonic_route_maps.yml]: I have made this change in all files (this one and all example files). Thanks for reminding me of the reason for requiring the use of '_' in Ansible argument names.

File: models/enterprise_sonic/route_maps/sonic_route_maps.yml (hunk 2 above)
Comment (kerry-meyer) [reply models/enterprise_sonic/route_maps/sonic_route_maps.yml]: I have made this change in all files (this one and all example files). Thanks for reminding me of the reason for requiring the use of '_' in Ansible argument names.

File: models/enterprise_sonic/route_maps/sonic_route_maps.yml (hunk 3 above)
Comment (kerry-meyer) [reply models/enterprise_sonic/route_maps/sonic_route_maps.yml]: Done (in all files, for all affected variable names).

**Comment by kerry-meyer** [comment]
I have fixed the problems identified in the first pass review comments. The corresponding changes have been made in all files of the change set (the ".yml" data model definition file and all "example" files).

**Comment by kerry-meyer** [comment]
I have deleted the "dictionary deletion" example case because it is not supported by the Ansible infrastructure.

There is no way for a resource module to distinguish a dictionary name (key) provided in a playbook with no value specified (the intended indicator for dictionary deletion) from any other dictionaries defined at the same nesting level in the argspec, but not present in the playbook. In both cases, Ansible provides the key for the dictionary, followed by a null value.

**Review by stalabi1 — APPROVED** [review]

**Review by stalabi1 — APPROVED** [review]

**Pull request merged**